core/src/execution/datafusion/expressions/regexp.rs

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use crate::{errors::CometError, execution::datafusion::expressions::utils::down_cast_any_ref};
+use arrow_array::{builder::BooleanBuilder, Array, RecordBatch, StringArray};
+use arrow_schema::{DataType, Schema};
+use datafusion::logical_expr::ColumnarValue;
+use datafusion_common::ScalarValue;
+use datafusion_physical_expr::PhysicalExpr;
+use regex::Regex;
+use std::{
+    any::Any,
+    fmt::{Display, Formatter},
+    hash::Hasher,
+    sync::Arc,
+};
+
+#[derive(Debug, Hash)]
+pub struct RLike {
+    child: Arc<dyn PhysicalExpr>,
+    pattern: Arc<dyn PhysicalExpr>,
+}
+
+impl RLike {
+    pub fn new(child: Arc<dyn PhysicalExpr>, pattern: Arc<dyn PhysicalExpr>) -> Self {
+        Self { child, pattern }
+    }
+}
+
+impl Display for RLike {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "RLike [child: {}, pattern: {}] ",
+            self.child, self.pattern
+        )
+    }
+}
+
+impl PartialEq<dyn Any> for RLike {
+    fn eq(&self, other: &dyn Any) -> bool {
+        down_cast_any_ref(other)
+            .downcast_ref::<Self>()
+            .map(|x| self.child.eq(&x.child) && self.pattern.eq(&x.pattern))
+            .unwrap_or(false)
+    }
+}
+
+impl PhysicalExpr for RLike {
```

Comment:
In `strings.rs` we have
```
// make_predicate_function!(RLike, rlike_dyn, rlike_utf8_scalar_dyn);
```
Is it possible to use it instead?

Reply:
Arrow does not provide an implementatin of `rlike_dyn` or `rlike_utf8_scalar_dyn`

Reply:
Also, I am expecting us to start specializing our implementation to make it compatible with Spark, so it seems better to have our own implementation.

Reply:
Got it. For other functions e.g. `like`, we contributed in `arrow-rs` https://github.com/apache/arrow-rs/pull/3502 in the past...

Reply:
I think it makes sense for Like. Regular expression support is just going to be quite challenging if we choose to try and mimic the behavior of the java regexp engine.